### ISSUE TITLE
Macro with name `on` unexpected behaviour

### ISSUE THREAD

**Issue by coffeepots** [nim-lang/Nim#7857]
Module A:
```
macro ona*(body: untyped): untyped =
  discard

macro on*(body: untyped): untyped =
  discard
```

Module B:
```
import modulea

ona:
  echo "a" # no problem

on:
  echo "b" # Error: expression 'echo ["b"]' has no type (or is ambiguous)
```

Changing the name of the macro from `on` fixes the problem.

**Comment by data-man** [comment]
For ```off``` the same.

**Comment by arnetheduck** [comment]
isn't this because https://github.com/nim-lang/Nim/blob/a39b989ce238546eca64e9c37734bc48f61e280c/lib/system.nim#L56?

**Comment by Araq** [comment]
Yeah, but I wonder why overloading resolution picks up ``consts``. :-)

**Comment by metagn** [comment]
It finds `system.on` first then calls `semIndirectOp` which calls `semOpAux` which typechecks all the arguments.

Maybe other symbols should also call `semDirectOp`/`semMagic` the same way [ambiguous type symbols](https://github.com/nim-lang/Nim/blob/4ef06a5cc56f6ef32b8ac1e7c11592597406486a/compiler/semexprs.nim#L3323) do
